samples/nrf9160/lwm2m_client/README.rst

```diff
@@ -1,219 +1,626 @@
+See the `server setup`_ intructions for how to do this.
 
-1. Open the Leshan Demo Server web UI.
-#. Click on :guilabel:`Security` in the upper-right corner.
-#. Click on :guilabel:`Add new client security configuration`.
-#. Enter the following data and click :guilabel:`Create`:
+Sensor simulation
+=================
+
+The sample allows choosing whether to use actual sensor measurements or
+simulated sensor data for all sensors (including the accelerometer).
+If the sample is running on the nRF9160 DK, only simulated sensor data is
+available, as it does not have any of the external sensors needed for actual
+measurements.
+
+Enable the :option:`CONFIG_<SENSOR_TYPE>_USE_SIM` configuration options for
+each sensor you want to receive simulated data from.
+
+.. _Notifications:
+
+Notifications
+=============
 
-    * Client endpoint - nrf-{your Device IMEI}
-    * Security mode - Pre-Shared Key
-    * Identity: - nrf-{your Device IMEI}
-    * Key - 000102030405060708090a0b0c0d0e0f
+LwM2M specifies the Notify operation, which can be used to notify the server of
+changes to the value of a resource field, e.g. the measured value of a
+temperature sensor.
+This allows active monitoring while using minimal bandwidth, as notifications
+can be sent with the updated values without the need for the server querying the
+client regularly.
 
-#. :ref:`Build and run the LwM2M Client sample <build_lwm2m>`.
+To enable notifications, the server must initiate an observation request on the
+resource(s) in question.
+See the `notifications setup`_ for how to do this.
+
+.. _Sensor Module:
+
+Sensor Module
+=============
+
+The sample has a sensor module that, if enabled, reads the selected sensors
+and updates the client's resource values if it detects a sufficiently large
+change in one of the values.
+The threshold for what is a sufficiently large change can be configured, and can
+e.g. be a change in temperature of 1 degree Celsius.
+
+Each sensor can be enabled separately.
+The sampling period and change threshold of that sensor can also be configured
+independently of all the other sensors.
+
+The sensor module is intended to be used together with `Notifications`_.
+If notifications are enable for a Sensor Value resource, and the corresponding
+sensor is enabled in the sensor module, a notification will be sent only when
+that value changes significantly (as specified by the change threshold).
+Thus the bandwidth usage can be significantly limited, while simultaneously
+catching important changes in sensor values.
+
+See the `sensor module options`_ for how to enable and
+configure the sensor module.
 
 Configuration
 *************
 
 |config|
 
+Setup
+=====
+
+.. _server setup:
+
+The following instructions describe how to register your device to
+`Leshan Demo Server`_ or `Coiote Device Management`_.
+
+.. tabs::
+
+   .. tab:: Leshan Demo Server
+
+      Since Leshan does not have a No Security option, DTLS must be
```

Comment:
This is not correct, if you don't want security you simply skip the security configuration. You don't need to register a device specifcally if no DTLS is used, you just connect.

Reply:
Fixed